R/data-combined.R

```diff
@@ -0,0 +1,680 @@
+                                      yOffsets = 0,
+                                      xScaleFactors = 1,
+                                      yScaleFactors = 1) {
+      # check that the arguments to parameters make sense
+      xOffsets <- validateVectorArgs(xOffsets, type = "numeric")
+      yOffsets <- validateVectorArgs(yOffsets, type = "numeric")
+      xScaleFactors <- validateVectorArgs(xScaleFactors, type = "numeric")
+      yScaleFactors <- validateVectorArgs(yScaleFactors, type = "numeric")
+      names <- validateVectorArgs(names, type = "character")
+
+      # apply specified data transformations
+      private$.dataCombined <- private$.dataTransform(
+        data          = private$.dataCombined,
+        names         = names,
+        xOffsets      = xOffsets,
+        yOffsets      = yOffsets,
+        xScaleFactors = xScaleFactors,
+        yScaleFactors = yScaleFactors
+      )
+
+      # update private field with transformation values
+      private$.dataTransformations <- private$.extractTransforms(private$.dataCombined)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    ## getter methods ---------------
+
+    #' @description
+    #'
+    #' A method to extract a dataframe of simulated and/or observed data
+    #' (depending on instances of which classes have been added to the object).
+    #'
+    #' Note that the order in which you enter different object doesn't matter
+    #' because the returned dataframe is arranged alphabetically by dataset
+    #' name.
+    #'
+    #' @return
+    #'
+    #' In the returned dataframe, the following columns will always be present:
+    #'
+    #' name - group - dataType - xValues - xDimension - xUnit - yValues -
+    #' yErrorValues - yDimension - yUnit - yErrorType - yErrorUnit - molWeight
+    #'
+    #' @note
+    #'
+    #' The molecular weight (in `molWeight` column) is in `g/mol` units.
+
+
+    toDataFrame = function() {
+      # It is deliberate that this "cleaning" happens every time the user calls
+      # this method. R6 has a reference semantics, i.e., the object will be
+      # modified in place and the combined data will be updated with it. But,
+      # for future updates, we do need to retain some information that is
+      # omitted during cleaning.
+      #
+      # For example, we remove offset and scale factor columns during cleaning,
+      # but they need to be retained in the private copy of the combined
+      # dataframe because the datasets that might be added in the future will
+      # have their own offsets and scale factors, which will need to be appended
+      # to the existing ones.
+      private$.cleanDF(private$.dataCombined)
+    },
+
+    ## print method -----------------
+
+    #' @description
+    #' Print the object to the console
+
+    print = function() {
+      # group map contains names and nature of the datasets and grouping details
+      private$printLine("DataCombined", addTab = FALSE)
```

Comment:
First line of `print` should be

`private$printClass()`

Then you don't need `printLine("DataCombined")`

Reply:
This is actually deliberate on my part. 

I'll double-check, but I am pretty sure that the alignment looked off if I used `private$printClass()` and this is why I used this workaround.

Reply:
Then `private$printClass()` should be fixed ;) We should use general functions if we already have them, instead of having "custom solutions" in every file.